docs/tutorials/excel-tutorial-create-custom-functions.md

```diff
@@ -3,8 +3,293 @@ title: Excel custom functions tutorial
+
+    This JSON describes the `stockPrice` function.
+
+    ```JSON
+    {
+        "id": "STOCKPRICE",
+        "name": "STOCKPRICE",
+        "description": "Fetches current stock price",
+        "helpUrl": "http://www.contoso.com/help",
+        "result": {
+            "type": "number",
+            "dimensionality": "scalar"
+        },  
+        "parameters": [
+            {
+                "name": "ticker",
+                "description": "stock ticker name",
+                "type": "string",
+                "dimensionality": "scalar"
+            }
+        ]
+    }
+    ```
+
+5. You must re-register the add-in in Excel in order for the new function to be available to end-users. Complete the following steps for the platform that you're using in this tutorial.
+
+    * If you're using Excel for Windows:
+
+        1. Close Excel and then reopen Excel.
+
+        2. In Excel, choose the **Insert** tab and then choose the down-arrow located to the right of **My Add-ins**.
+            ![Insert ribbon in Excel for Windows with the My Add-ins arrow highlighted](../images/excel-cf-register-add-in-1b.png)
+
+        3. In the list of available add-ins, find the **Developer Add-ins** section and select the **stock-ticker** add-in to register it.
+            ![Insert ribbon in Excel for Windows with the Excel Custom Functions add-in highlighted in the My Add-ins list](../images/excel-cf-register-add-in-2.png)
+
+    * If you're using Excel Online:
+
+        1. In Excel Online, choose the **Insert** tab and then choose **Add-ins**.
+            ![Insert ribbon in Excel Online with the My Add-ins icon highlighted](../images/excel-cf-online-register-add-in-1.png)
+
+        2. Choose **Manage My Add-ins** and select **Upload My Add-in**. 
+
+        3. Choose **Browse...** and navigate to the root directory of the project that the Yeoman generator created. 
+
+        4. Select the file **manifest.xml** and choose **Open**, then choose **Upload**.
+
+6. Now, let's try out the new function. In cell **B1**, type the text `=CONTOSO.STOCKPRICE("MSFT")` and press enter. You should see that the result in cell **B1** is the current stock price for one share of Microsoft stock.
+
+## Create a streaming asynchronous custom function
+
+The `stockPrice` function that you just created returns the price of a stock at a specific moment in time, but stock prices are always changing. Let's create a custom function that streams data from an API to get real-time updates on a stock price.
+
+Complete the following steps to create a custom function named `stockPriceStream` that requests the price of the specified stock every 1000 milliseconds (provided that the previous request has completed). While the initial request is in-progress, you may see the placeholder value **#GETTING_DATA** the cell where the function is being called. When a value is returned by the function, **#GETTING_DATA** will be replaced by that value in the cell.
```

Comment:
Nit: "When a value is returned by the function, **#GETTING_DATA** will be replaced by that value in the cell." -> "When a value is returned by the function, **#GETTING_DATA** is replaced by that value in the cell."